packages/ns-api/files/ns.reverseproxy

```diff
@@ -0,0 +1,331 @@
+
+    return location_id
+
+
+def __certificate_list():
+    cert_path = '/etc/acme'
+    certificate_list = os.listdir(cert_path)
+    cert_files = [file.removesuffix('.crt') for file in certificate_list if file.endswith('.crt')]
+    key_files = [file.removesuffix('.key') for file in certificate_list if file.endswith('.key')]
+    certificates = {'_lan': '/etc/nginx/conf.d/_lan'}
+    for cert_file in cert_files:
+        if cert_file in key_files:
+            certificates[cert_file.rsplit('/')[0]] = f'{cert_path}/{cert_file}'
+
+    return certificates
+
+
+cmd = sys.argv[1]
+
+if cmd == 'list':
+    print(json.dumps({
+        'list-certificates': {},
+        'add-path': {
+            'path': 'str',
+            'destination': 'str',
+            'description': 'str',
+            'allow': ['str']
+        },
+        'add-domain': {
+            'domain': 'str',
+            'certificate': 'str',
+            'destination': 'str',
+            'description': 'str',
+            'allow': ['str']
+        },
+        'delete-proxy': {
+            'id': 'str'
+        },
+        'list-proxies': {},
+        'edit-path': {
+            'id': 'str',
+            'path': 'str',
+            'destination': 'str',
+            'description': 'str',
+            'allow': ['str']
+        },
+        'edit-domain': {
+            'id': 'str',
+            'domain': 'str',
+            'certificate': 'str',
+            'destination': 'str',
+            'description': 'str',
+            'allow': ['str']
+        },
+        'check-config': {}
+    }))
+elif cmd == 'call':
+    action = sys.argv[2]
+    if action == 'list-certificates':
+        print(json.dumps({'values': list(__certificate_list().keys())}))
+    else:
+        e_uci = EUci()
+        try:
+            if action == 'add-path':
+                paths = [entry['location'] for entry in utils.get_all_by_type(e_uci, 'nginx', 'location').values()
+                         if entry['uci_server'] == '_lan']
+                # load and validate data
+                data = json.loads(sys.stdin.read())
+                if 'path' not in data:
+                    raise ValidationError('path', 'required')
+                if data['path'] in paths:
+                    raise ValidationError('path', 'unique', data['path'])
+                if 'destination' not in data:
+                    raise ValidationError('destination', 'required')
+                if not data['destination'].startswith('http://') and not data['destination'].startswith('https://'):
+                    raise ValidationError('destination', 'invalid', data['destination'])
+                if 'description' not in data or data['description'] == '':
+                    raise ValidationError('description', 'required')
+                if 'allow' in data and not isinstance(data['allow'], list):
+                    raise ValidationError('allow', 'invalid', data['allow'])
```

Comment:
Should we also add the validation to avoid the same path?

Reply:
Already there! [Just a few lines above](https://github.com/NethServer/nethsecurity/pull/227/files#diff-1a04a0ae222beb786726104911ef0bc7a5097be32a40699b9bc70382400cbdaeR103) :smile:

Reply:
Sorry I didn't spot it. :+1: